test/specs/errors/tests/09.modifier_error.js

```diff
@@ -0,0 +1,164 @@
+        var facetObj2 = {
+            "and":[
+                {"source":"accession", "choices":["2"]},
+                {"source":"*", "search":["test2"]}
+            ]
+        };
+
+
+        var baseUri = options.url + "/catalog/" + catalogId + "/entity/" + schemaName + ":" + tableName;
+        var facetError = "Given encoded string for facets is not valid.";
+        var location, uri;
+        var invalidPageConditionErrorObj = {
+                       'status': 'Invalid Page Criteria',
+                       'messageWithCondition': "Invalid uri: " + options.url + "/catalog/"+catalogId+"/entity/"+schemaName+":"+tableName+"/id=269@after(). Sort modifier is required with paging.",
+                       'messageWithConditions': "Invalid uri: " + options.url + "/catalog/"+catalogId+"/entity/"+schemaName+":"+tableName+"/id=269@after(3)@before(7). Sort modifier is required with paging.",
+                       'errorData': { 'redirectPath': schemaName+":"+tableName+'/id=269' }
+       };
+        var invalidFilterOperatorErrorObj = {
+                       'status': 'Invalid Filter',
+                       'message': "Invalid uri: " + options.url + "/catalog/"+catalogId+"/entity/"+schemaName+":"+tableName+"/id::gt:269. Couldn't parse 'id::gt:269' filter.",
+                       'errorData': { 'redirectPath':schemaName+":"+tableName+'/' },
+                       'messageWithSort': "Invalid uri: " + options.url + "/catalog/"+catalogId+"/entity/"+schemaName+":"+tableName+"/id::gt:269@sort(). Couldn't parse 'id::gt:269' filter.",
+                       'errorDataWithSort': { 'redirectPath': schemaName+":"+tableName+'/@sort()' }
+       };
+        var invalidFacetFilterErrorObj = {
+                       'status': 'Invalid Facet Filters',
+                       'message': "Given encoded string for facets is not valid.",
+                       'errorData': { 'redirectPath': schemaName+":"+tableName+'/@sort()' },
+                       'errorDataWithoutSort': { 'redirectPath': schemaName+":"+tableName+"/" }
+      };
+
+      var invalidSortCriteriaErrorObj = {
+                     'status': 'Invalid Sort Criteria',
+                     'message': "Column id is not sortable.",
+                     'errorData': { 'redirectPath': schemaName + ":" + tableName }
+
+    };
+
+      beforeAll(function (done) {
+        options.ermRest.resolve(baseUri + "@sort(id)", {cid:"test"}).then(function(response) {
+          references = response;
+          done();
+        });
+      });
+
+        describe("when uri doesn't have any facets, ", function() {
+            it("Location.facets should be undefined.", function() {
+                location = options.ermRest.parse(baseUri);
+                expect(location.facets).toBeUndefined();
+            });
+        });
+
+        describe("when uri have invalid facet blob", function() {
+            it("parser should throw an error.", function() {
```

Comment:
This is not needed as you are testing the same thing in depth few lines further.